### ISSUE TITLE
Ambiguous unit "1/min"

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1807]
**Reported by beutlich on 26 Oct 2015 13:55 UTC**
ITI unfortunately needs to provide an adapted MSL along with SimulationX. One major change is the replacement of the unit string "1/min" to "rpm" in all occurrences of
 * Modelica.SIunits
 * Modelica.Blocks.Math
 * Modelica.Electrical.Machines
 * Modelica.Fluid.Machines
 * Modelica.Magnetic.FundamentalWave
 * Modelica.Thermal.FluidHeatFlow
if the variable type or the quantity attribute is not set to Modelica.SIunits.AngularVelocity.

In the MSL (after 4fd2be8d1a291f0e9d6e1bfceff3a16e6b0b1dd0, 54fea525af2936fd665339cbbee055ed1a642ff5) the quantityless unit string "1/min" has to be interpreted as the unit of the Angular Velocity where 1/min = rpm = r/min = U/min = rev/min = pi/30 s<sup></sup>-1.

However SimulationX considers the quantityless unit string "1/min" as the unit of the Frequency, i.e., 1/min = 1/60 s<sup></sup>-1 = 1/60 Hz.

This conflicts with the requirement from the MSL. Unfortunatly we must not change this unit mapping in order to not break the backwards-compatibility with existing user models.

I propose to revert 4fd2be8d1a291f0e9d6e1bfceff3a16e6b0b1dd0, 54fea525af2936fd665339cbbee055ed1a642ff5 in order to not choose the ambiguous unit string "1/min" for the Angular Velocity. Instead of the previous unit string "r/min" either the unit string "rpm" or "rev/min" could be chosen.

Please see the updated implementations To_rpm2 and To_rpm3 that fix Modelica.Blocks.Math.UnitConversions.To_rpm accordingly.
```mo
model T1807 "Adapted blocks as replacement of Modelica.Blocks.Math.UnitConversions.To_rpm"
  import Modelica.Constants.pi;

  block To_rpm "Convert from radian per second to revolutions per minute"
    // Same as Modelica.Blocks.Math.UnitConversions.To_rpm
    extends Modelica.Blocks.Interfaces.PartialConversionBlock(u(unit="rad/s"), y(unit="1/min"));
  equation
    y = Modelica.SIunits.Conversions.to_rpm(u);
  end To_rpm;

  block To_rpm2 "Convert from radian per second to revolutions per minute"
    // y.unit changed to "rpm" (as found in the SimulationX specific MSL)
    extends Modelica.Blocks.Interfaces.PartialConversionBlock(u(unit="rad/s"), y(unit="rpm"));
  equation
    y = Modelica.SIunits.Conversions.to_rpm(u);
  end To_rpm2;

  block To_rpm3 "Convert from radian per second to revolutions per minute"
    // typeof(y) changed to AngularVelocity_rpm
    Modelica.Blocks.Interfaces.RealInput u(unit="rad/s") "Connector of Real input signal to be converted" annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
    Modelica.SIunits.Conversions.NonSIunits.AngularVelocity_rpm y(displayUnit="rpm") "Connector of Real output signal containing input signal u in another unit" annotation (Placement(transformation(extent={{100,-10},{120,10}})));
  equation
    y = Modelica.SIunits.Conversions.to_rpm(u);
  end To_rpm3;

  Modelica.Blocks.Sources.Constant const(k=pi/30) annotation(Placement(transformation(extent={{-105,15},{-85,35}})));
  To_rpm to_rpm1 annotation(Placement(transformation(extent={{-35,35},{-15,55}})));
  To_rpm2 to_rpm2 annotation(Placement(transformation(extent={{-35,0},{-15,20}})));
  To_rpm3 to_rpm3 annotation(Placement(transformation(extent={{-35,-40},{-15,-20}})));
  equation
    connect(const.y,to_rpm1.u) annotation(Line(points={{-84,25},{-79,25},{-42,25},{-42,45},{-37,45}}, color={0,0,127}, thickness=0.0625));
    connect(to_rpm2.u,const.y) annotation(Line(points={{-37,10},{-42,10},{-79,10},{-79,25},{-84,25}}, color={0,0,127}, thickness=0.0625));
    connect(to_rpm3.u,const.y) annotation(Line(points={{-37,-30},{-42,-30},{-79,-30},{-79,25},{-84,25}}, color={0,0,127}, thickness=0.0625));
  annotation(uses(Modelica));
end T1807;
```

The unit strings "1/s" is not affected since it is considered as a unit of the Frequency both in the MSL and SimulationX.

The unit string "1/h" does not occur in the current MSL and is also not affected.

P.S. I am aware of the 2007 discussions from https://groups.google.com/forum/#!msg/modelica-design/vYoCue9TyUU/AzNtziMNmM4J which lead to 4fd2be8d1a291f0e9d6e1bfceff3a16e6b0b1dd0, 54fea525af2936fd665339cbbee055ed1a642ff5.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1807

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 26 Oct 2015 14:45 UTC**
This is the Dymola-warning-free model T1807 where "rpm" is now "rev/min" and previous To_rpm (part of MSL) no longer gives a warning about not matching units.
```mo
model T1807 "Adapted blocks as replacement of Modelica.Blocks.Math.UnitConversions.To_rpm"
  import Modelica.Constants.pi;

  block To_rpm "Convert from radian per second to revolutions per minute"
    // avoid unit conversion warning in Dymola by introducing onerev
    extends Modelica.Blocks.Interfaces.PartialConversionBlock(u(unit="rad/s"), y(unit="1/min"));
  protected
    final parameter Real onerev(unit="rev") = 1.0;
  equation
    y = Modelica.SIunits.Conversions.to_rpm(u)/onerev;
  end To_rpm;

  block To_rpm2 "Convert from radian per second to revolutions per minute"
    // y.unit changed to "rev/min"
    extends Modelica.Blocks.Interfaces.PartialConversionBlock(u(unit="rad/s"), y(unit="rev/min"));
  equation
    y = Modelica.SIunits.Conversions.to_rpm(u);
  end To_rpm2;

  block To_rpm3 "Convert from radian per second to revolutions per minute"
    // typeof(y) changed to AngularVelocity_rpm
    Modelica.Blocks.Interfaces.RealInput u(unit="rad/s") "Connector of Real input signal to be converted" annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
    Modelica.SIunits.Conversions.NonSIunits.AngularVelocity_rpm y(displayUnit="rev/min") "Connector of Real output signal containing input signal u in another unit" annotation (Placement(transformation(extent={{100,-10},{120,10}})));
  equation
    y = Modelica.SIunits.Conversions.to_rpm(u);
  end To_rpm3;

  Modelica.Blocks.Sources.Constant const(k=pi/30) annotation(Placement(transformation(extent={{-105,15},{-85,35}})));
  To_rpm to_rpm1 annotation(Placement(transformation(extent={{-35,35},{-15,55}})));
  To_rpm2 to_rpm2 annotation(Placement(transformation(extent={{-35,0},{-15,20}})));
  To_rpm3 to_rpm3 annotation(Placement(transformation(extent={{-35,-40},{-15,-20}})));
  equation
    connect(const.y,to_rpm1.u) annotation(Line(points={{-84,25},{-79,25},{-42,25},{-42,45},{-37,45}}, color={0,0,127}, thickness=0.0625));
    connect(to_rpm2.u,const.y) annotation(Line(points={{-37,10},{-42,10},{-79,10},{-79,25},{-84,25}}, color={0,0,127}, thickness=0.0625));
    connect(to_rpm3.u,const.y) annotation(Line(points={{-37,-30},{-42,-30},{-79,-30},{-79,25},{-84,25}}, color={0,0,127}, thickness=0.0625));
  annotation(uses(Modelica));
end T1807;
```

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 10 Dec 2015 14:44 UTC**
Fixed in 010357042274f50771da9998b529b9fa0b06b48e by replacing "1/min" with "rev/min" in package Modelica and ModelicaTest.

Reasoning:
- AngularVelocity has unit "rad/s", Frequency has unit "Hz". So in these two cases the units are clearly distinguishd (both are SIunits "1/s"), and the units are totally clear for the end user (which would not be the case if in one, or both cases "1/s" would be used).
- In technical systems the unit "revolutions per minute" is used very extensively and we need to support it in the library. We made a wrong decision previously, because there will be many people that will not at once understand the unit "1/min" (it doesn't matter whether this is clear for some people or there is general agreement; the important point is whether the people working with it, understand it). Doing it in a similar way as with "rad/s" and "Hz", we need a unique name. Since some tools (at least SimulationX and Dymola) support "rev" as unit (as they support "rad"), it seems most practical to use "rev/min" (and then the unit should be clear for most people).

Note, even when using SIunits, unit handling is ambiguous and one needs to be pragmatic and rely on heuristics that is widely accepted.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 10 Dec 2015 14:44 UTC**
Changed the ambiguous unit "1/min" to "rev/min"
